spec/lang/intrinsic.md

```diff
@@ -194,21 +194,33 @@ impl<M: Memory> Machine<M> {
         };
 
         let func = self.fn_from_addr(ptr.addr)?;
+        if func.args.len() != 1 {
+            throw_ub!("invalid first argument to `Intrinsic::Spawn`, function should take one argument.");
+        }
 
-        if func.args.len() != 0 {
-            throw_ub!("invalid first argument to `Intrinsic::Spawn`, function takes arguments");
+        let (data_ptr, data_ptr_ty) = arguments[1];
+        if !matches!(data_ptr_ty, Type::Ptr(PtrType::Raw)) {
+            throw_ub!("invalid second argument to `Intrinsic::Spawn`, data pointer should be a raw pointer.");
+        }
+        // This logic is taken from `eval(Terminator::Call)` and `eval_argument`.
```

Comment:
```suggestion
        // This logic is taken from `eval(Terminator::Call)` and `eval_argument`.
        // FIXME: find a way to avoid the code duplication.
```